server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -375,10 +380,19 @@ public static DFSLoggerInputStreams readHeaderAndReturnStream(FSDataInputStream
         log.debug("Using {} for decrypting WAL", cryptoService.getClass().getSimpleName());
         decryptingInput = cryptoService instanceof NoCryptoService ? input
             : new DataInputStream(decrypter.decryptStream(input));
-      } else {
-        log.error("Unsupported WAL version.");
-        input.seek(0);
+      } else if (Arrays.equals(magicBuffer, magic3)) {
+        // Read logs files from Accumulo 1.9
+        String cryptoModuleClassname = input.readUTF();
+        if (!cryptoModuleClassname.equals("NullCryptoModule")) {
+          throw new IllegalArgumentException(
+              "Old encryption modules not supported at this time.  Unsupported module : "
+                  + cryptoModuleClassname);
+        }
+
         decryptingInput = input;
+      } else {
+        throw new IllegalArgumentException(
```

Comment:
This will catch immediately in the catch block below, and print the incorrect message about getting an EOFException (which isn't true in this case). This is probably doing the right thing, but the catch block should be corrected to match.

Reply:
Good catch.  I am going to change it to catch EOFException, which seems like the right thing to do.  The change could cause it fail in some cases where it used to succeed.  However the current code could succeed when it should fail.

Reply:
I think that would be misleading to a user debugging the failure. The end of file hasn't been reached. Rather, the content in the file is unexpected. I think it would be better to simply update the message in the catch block, since there is more than one exception which could be caught there.

Reply:
@ctubbsii  is your last comment about the code before or after b3f9d45 ?

Reply:
I misunderstood that you were going to throw EOFException, not catch it. Catching the more specific error makes sense to me... but we should take care that the RTE that will now fall through is handled (or propagated) appropriately by the caller(s).

From a quick examination, it looks like the callers are also logging... so you should probably just include whatever message stuff you want in the thrown exception, instead of logging first, and then throwing it, where it will be logged again, and then thrown again... etc.